src/daft-io/src/s3_like.rs

```diff
@@ -693,55 +670,47 @@ impl S3LikeSource {
                 )),
             };
 
-            let response = if self.anonymous {
-                request
-                    .customize_middleware()
-                    .await
-                    .unwrap()
-                    .map_operation::<Error>(|mut o| {
-                        {
-                            let mut properties = o.properties_mut();
-                            #[allow(unused_mut)]
-                            let mut config = properties
-                                .get_mut::<::aws_sig_auth::signer::OperationSigningConfig>()
-                                .expect("signing config added by make_operation()");
-
-                            config.signing_requirements = SigningRequirements::Disabled;
-                        }
-                        Ok(o)
-                    })
-                    .unwrap()
-                    .send()
-                    .await
-            } else {
-                request.send().await
-            };
+            let response = request.send().await;
 
             match response {
                 Ok(v) => {
                     let body = v.body;
-                    let owned_string = uri.to_owned();
-                    let stream = body
-                        .map_err(move |e| {
-                            UnableToReadBytesSnafu {
-                                path: owned_string.clone(),
-                            }
-                            .into_error(e)
-                            .into()
-                        })
-                        .boxed();
+                    struct FuturesStreamCompatByteStream {
```

Comment:
Why do we need to do this?

Reply:
This is needed because AWS removed the dependency of the SDK on `futures`and instead implemented their own `ByteStream` type for `body`. So we need to just make a small wrapper that implements `Stream` to make it a shape that matches the trait method signature